elasticsearch_dsl/search.py

```diff
@@ -587,6 +605,9 @@ def to_dict(self, count=False, **kwargs):
             if self._script_fields:
                 d['script_fields'] = self._script_fields
 
+            if self._rescore:
+                d['rescore'] = self._rescore
```

Comment:
Can you also add complementary logic to `update_from_dict` (plus related test) here? Thanks!